cbt/src/CBT/Image/BuildDefinition.hs

```diff
@@ -87,43 +90,39 @@ fromDockerfileContent imageName content =
       . Text.encodeUtf8
       $ toText content
 
-hashContentTag :: MonadUnliftIO m => [FilePath] -> m CBT.Image.Tag
+hashContentTag :: MonadUnliftIO m => [HashItem] -> m CBT.Image.Tag
 hashContentTag = fmap (CBT.Image.Tag . convertText . show) . runResourceT . hashContents
 
-hashContents :: forall m . MonadUnliftIO m => [FilePath] -> m (Hash.Digest Hash.SHA3_256)
+hashContents :: forall m . MonadUnliftIO m => [HashItem] -> m (Hash.Digest Hash.SHA3_256)
 hashContents contents = runResourceT $ Hash.hashFinalize <$> runConduit conduit
   where
     conduit
       =  Conduit.sourceList contents
       .| expand
-      .| readFile
+      .| Conduit.awaitForever (\chunk -> Conduit.yield "0" >> Conduit.yield chunk)
```

Comment:
That "0" byte (literally any byte would do I'm aware its not a 0 byte!) is needed to avoid hashing different inputs to the same output if the chunks came in one time as:

```
aaa
bbb
```

and the next time as:

```
aa
abbb
```

which together *still* produces the string `aaabbb` while being different inputs. We need to have a delimiter to make these different hashed strings.